transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java

```diff
@@ -40,12 +40,12 @@
     private volatile InetSocketAddress local;
     private volatile InetSocketAddress remote;
 
-    EpollSocketChannel(Channel parent, int fd) {
+    EpollSocketChannel(Channel parent, int fd, InetSocketAddress remote) {
         super(parent, fd);
         config = new EpollSocketChannelConfig(this);
         // Directly cache the remote and local addresses
         // See https://github.com/netty/netty/issues/2359
-        remote = Native.remoteAddress(fd);
+        this.remote = remote;
         local = Native.localAddress(fd);
```

Comment:
Should we add a comment here (or somewhere) that a race condition exists for the local address?  Also indicating that it can't be avoided because the `accept` interface doesn't return the remote address.